source/configuration_manual/stats/index.rst

```diff
@@ -77,6 +92,102 @@ Note: While the top level metrics (e.g., ``imap_command`` above) are
+maximum magnitude, and the base.  The exact configuration syntax is:
+``field:exponential:min:max:base``
+
+Note: Currently, only base 2 and base 10 are supported.
+
+The first range starts at negative infinity and ends at ``pow(base, min)``.
+The second range begins at ``pow(base, min) + 1`` and ends at
+``pow(base, min + 1)``, the next covers ``pow(base, min + 1) + 1`` to
+``pow(base, min + 2)``, and so on.  The last range covers
+``pow(base, max) + 1`` to positive infinity.
+
+For example, given the specification ``duration:exponential:1:5:10``, the
+ranges would be:
+
+* (-inf, 10]
+* [11, 100]
+* [101, 1000]
+* [1001, 10000]
+* [10001, 100000]
+* [100001, +inf)
+
+Much like the metric names generated with the ``discrete`` aggregation
+function, the ones generated by the ``exponential`` function include
+information about the value of the field.  However, in this case it is the
+range the value belongs to.
+
+Specifically, it is the name of the field being quantized, and the lower and
+upper bounds for the range.
+
+Example::
+
+   metric imap_command {
+     event_name = imap_command_finished
+     group_by = cmd_name duration:exponential:1:5:10
+   }
+
+This will generate metric names of the format
+``imap_command_{cmd}_duration_{min}_{max}`` where ``{cmd}`` is the IMAP
+command name, and ``{min}`` and ``{max}`` are the range bounds.  Therefore,
+for a ``SELECT`` IMAP command, the possible generated metric names are:
+
+* ``imap_command_select_ninf_10``
+* ``imap_command_select_11_100``
+* ``imap_command_select_101_1000``
+* ``imap_command_select_1001_10000``
+* ``imap_command_select_10001_100000``
+* ``imap_command_select_100001_inf``
+
+Note: Since the metric names cannot contain -, the string ``ninf`` is used
+to denote negative infinity.
+
+Note: Much like in the ``discrete`` case, the metrics are allocated only
+when first observed.
+
+Finally, because all intermediate level metrics are generated as well.  The
+above example, will also generate all of the following metrics:
+
+ - ``imap_command``
+ - ``imap_command_select``
+
+`linear`
+--------
+
+The field's integer value is quantized into linearly sized ranges.
+
+The linear aggregation function takes three colon delimited integer
+arguments that define the set of ranges used: the minimum value, the
+maximum value, and the range step size.  The exact configuration syntax is:
+``field:linear:min:max:step``
+
+The first range starts at negative infinity and ends at ``min``.  The second
+range begins at ``min + 1`` and ends at ``min + step``, the next covers
+``min + step + 1`` to ``min + (2 * step)``, and so on.  The last range
+covers ``max + 1`` to positive infinity.
+
+For example, given the specification ``bytes_out:linear:0:5000:1000``, the
+ranges would be:
+
+* (-inf, 0]
+* [1, 1000]
+* [1001, 2000]
+* [2001, 3000]
+* [3001, 4000]
+* [4001, 5000]
+* [5001, +inf)
+
+See the description of the ``exponential`` aggregation function for how
```

Comment:
Consider adding a link here, in case linear and exponential are ever turned into separate pages or separated by more content?

Reply:
Makes sense.